homeassistant/components/kodi/browse_media.py

```diff
@@ -161,55 +155,94 @@ def item_payload(item, media_library):
 
     Used by async_browse_media.
     """
-    if "songid" in item:
-        media_content_type = MEDIA_TYPE_TRACK
-        media_content_id = f"{item['songid']}"
-    elif "albumid" in item:
-        media_content_type = MEDIA_TYPE_ALBUM
-        media_content_id = f"{item['albumid']}"
-    elif "artistid" in item:
-        media_content_type = MEDIA_TYPE_ARTIST
-        media_content_id = f"{item['artistid']}"
-    elif "movieid" in item:
-        media_content_type = MEDIA_TYPE_MOVIE
-        media_content_id = f"{item['movieid']}"
-    elif "episodeid" in item:
-        media_content_type = MEDIA_TYPE_EPISODE
-        media_content_id = f"{item['episodeid']}"
-    elif "seasonid" in item:
-        media_content_type = MEDIA_TYPE_SEASON
-        media_content_id = f"{item['tvshowid']}/{item['season']}"
-    elif "tvshowid" in item:
-        media_content_type = MEDIA_TYPE_TVSHOW
-        media_content_id = f"{item['tvshowid']}"
-    else:
-        # this case is for the top folder of each type
-        # possible content types: album, artist, movie, library_music, tvshow
-        media_content_type = item.get("type")
-        media_content_id = ""
-
     title = item["label"]
-    can_play = media_content_type in PLAYABLE_MEDIA_TYPES and media_content_id
-    can_expand = media_content_type in EXPANDABLE_MEDIA_TYPES
 
     thumbnail = item.get("thumbnail")
     if thumbnail:
         thumbnail = media_library.thumbnail_url(thumbnail)
 
-    if media_content_type == MEDIA_TYPE_MOVIE and not media_content_id:
-        media_class = MEDIA_CLASS_DIRECTORY
-        can_expand = True
-    else:
-        media_class = CONTENT_TYPE_MEDIA_CLASS[media_content_type]
+    if "songid" in item:
+        return BrowseMedia(
+            title=title,
+            media_class=CONTENT_TYPE_MEDIA_CLASS[MEDIA_TYPE_TRACK],
```

Comment:
If we think we don't know all the media types out there yet, we should consider making the media class lookup robust by handling the KeyError. Like this:
https://github.com/home-assistant/core/pull/39841

Reply:
Done